e2e/e2e-create-windows-nodepool.sh

```diff
@@ -7,24 +7,34 @@ source e2e-helper.sh
 log "Starting to create windows nodepool"
 
 if [[ "$RESOURCE_GROUP_NAME" == *"windows"*  ]]; then
-    RESOURCE_GROUP_NAME="$RESOURCE_GROUP_NAME"-"$WINDOWS_E2E_IMAGE"
+    RESOURCE_GROUP_NAME="$RESOURCE_GROUP_NAME"-"$WINDOWS_E2E_IMAGE"-v2
 fi
 
 out=$(az aks nodepool list --cluster-name $CLUSTER_NAME -g $RESOURCE_GROUP_NAME | jq '.[].name')
 
 if [[ "$out" != *"winnp"* ]]; then
     log "Creating windows nodepool"
-    if [ "$WINDOWS_E2E_IMAGE" == "2019-containerd" ]; then
-        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --node-count 1
-    elif [ "$WINDOWS_E2E_IMAGE" == "2022-containerd" ]; then
-        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --os-sku Windows2022 --node-vm-size Standard_D2_v2 --node-count 1
-    elif [ "$WINDOWS_E2E_IMAGE" == "2022-containerd-gen2" ]; then
-        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --os-sku Windows2022 --node-vm-size Standard_DS2_v2 --node-count 1
+    retval=0
+    az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --os-sku $WINDOWS_E2E_OSSKU --node-vm-size $WINDOWS_E2E_VMSIZE --node-count 1 || retval=$?
+    if [ "$retval" -ne 0 ]; then
+        log "Other pipeline may be creating the same nodepool, waiting for ready"
     else
-        err "Invalid windows e2e image version"
-        exit 1
+        log "Created windows nodepool"
     fi
-    log "Created windows nodepool"
 else
     log "Already create windows nodepool"
 fi
+
+provisioning_state=$(az aks nodepool show --cluster-name $CLUSTER_NAME -g $RESOURCE_GROUP_NAME -n "winnp" -ojson | jq '.provisioningState' | tr -d "\"")
+if [ "$provisioning_state" == "Creating" ]; then
+    log "Other pipeline may be creating the same nodepool, waiting for ready"
+    az aks nodepool wait --nodepool-name "winnp" --cluster-name $CLUSTER_NAME --resource-group $RESOURCE_GROUP_NAME --created --interval 60 --timeout 1800
+fi
+
+provisioning_state=$(az aks nodepool show --cluster-name $CLUSTER_NAME -g $RESOURCE_GROUP_NAME -n "winnp" -ojson | jq '.provisioningState' | tr -d "\"")
+if [ "$provisioning_state" == "Succeeded" ]; then
+    log "Windows nodepool is in succeed state"
+else
+    err "Windows nodepool is not in succeeded state"
```

Comment:
Could we log the unexpected state for troubleshooting?

Reply:
Sure. Will add it.